### ISSUE TITLE
关于labels的问题

### ISSUE THREAD

**Issue by lzy-ps** [wdndev/tiny-llm-zh#10]
PTMDataset里get_item
X = np.array(sample[:-1]).astype(np.int64)
Y = np.array(sample[1:]).astype(np.int64)
input_ids = torch.LongTensor(X)
labels = torch.LongTensor(Y)
labels不是已经右移了。
那为什么在模型前向传播的时候又右移
shift_logits = logits[..., :-1, :].contiguous()
shift_labels = labels[..., 1:].contiguous()

**Comment by wdndev** [comment]
确实，真的有点重复了，出现大问题了

**Comment by lzy-ps** [comment]
> 你是对的，我这面ptm的dataload类有问题，label应该是input的copy，不用偏移，我修正一下

好的。我还有个问题想问您，就是在dpo的时候是不是模型太小放开全部参数进行训练是不是不太好，感觉很容易训崩。

**Comment by wdndev** [comment]
不用这么客气。是的，你说的这个问题是存在的，很容易训练崩掉。我训练了好多次，正负例的gap才拉大一点。不过这个主要是用来学习的项目。效果不是第一位，能跑的条件下，尽可能效果好。
